src/metabase/search/appdb/specialization/mysql.clj

```diff
@@ -0,0 +1,73 @@
+(ns metabase.search.appdb.specialization.mysql
+  (:require
+   [clojure.math :as math]
+   [clojure.string :as str]
+   [honey.sql :as sql]
+   [metabase.search.appdb.specialization.api :as specialization]
+   [toucan2.core :as t2]))
+
+(defmethod specialization/table-schema :mysql [base-schema]
+  (into [[:id :bigint :auto-increment :primary-key]
+         [:search_text :text :not-null]
+         [:with_native_query_text :text :not-null]]
+        (->> base-schema
+             (map (fn [cdef] (if (some #{:timestamp-with-time-zone} cdef)
+                               (mapv #(if (= % :timestamp-with-time-zone) :timestamp %) cdef)
+                               cdef))))))
```

Comment:
nit: I think this will read much better with a `for` expression.

At a higher level, if the implicit to-and-from conversion to UTC is OK, why don't we just do that for all the databases?

If we do need to do this, I think it would be nice to abstract this out to a `(replace xs old-value new-value)` helper for readability. And do we really need the `some` optimization if the whole index creation process is dominated by IO?